doc/source/release-notes/Percona-XtraDB-Cluster-5.6.27-25.13.rst

```diff
@@ -0,0 +1,78 @@
+
+* Fixed error when changing ``wsrep_provider`` in middle of a transaction or as part of procedure/trigger. This is now blocked to avoid inconsistency.
+
+* Removed support for ``innodb_fake_changes`` variable.
+
+* Fixed TOI state inconsistency caused by ``DELAYED_INSERT`` on MyISAM table (``TOI_END`` was not called). Now the ``DELAYED_`` qualifier will be ignored and statement will be interpreted as normal ``INSERT``.
+
+* Corrected locking semantics with FTWRL. It now avoids freeing inheritted lock if follow-up ``FLUSH TABLE`` statement fails. Only frees self-acquired lock.
+
+* ``wsrep_on`` is now a session only variable. That means toggling it will not affect other clients connected to said node. Only the session/client modifying it will be affected. Trying to toggle ``wsrep_on`` in the middle of a transaction will now result in an error. Trx will capture the state of ``wsrep_on`` during start and will continue to use it. Start here means when the first logical changing statement is executed within transaction context.
+
+* Fixed crash caused by ``GET_LOCK`` + ``drupal_workaround``. ``GET_LOCK`` path failed to free all instances of user-level locks after it inherited ``multiple-user-locks`` from PS. The cleanup code now removes all possible references of locks.
+
+* Corrected DDL and DML semantics for MyISAM:
+
+  * DDL (CREATE/DROP/TRUNCATE) on MyISAM will be replicated irrespective of ``wsrep_replicate_miysam`` value
+  * DML (INSERT/UPDATE/DELETE) on MyISAM will be replicated only if ``wsrep_replicate_myisam`` is enabled
+  * SST will get full transfer irrespective of ``wsrep_replicate_myisam`` value (it will get MyISAM tables from donor if any)
+  * Difference in configuration of ``pxc-cluster`` node on ``enforce_storage_engine`` front may result in picking up different engine for same table on different nodes
+  * CTAS statements use non-TOI replication and are replicated only if there is involvement of InnoDB table that needs trx (involvement of MyISAM table will cause CTAS statement to skip replication)
+
+Known Bugs
+==========
+
+* `1443755 <https://bugs.launchpad.net/percona-xtradb-cluster/+bug/1443755>`_: Causal reads introduces surprising latency in single node clusters.
+
+* `1522385 <https://bugs.launchpad.net/percona-xtradb-cluster/+bug/1522385>`_: Holes are introduced in Master-Slave GTID eco-system on replicated nodes if any of the cluster nodes are acting as asynchronous slaves to an independent master.
+
+* SST fails with ``innodb_data_home_dir``/``innodb_log_home_dir``. This is a bug in Percona XtraBackup. It should be fixed in the next 2.3.2 release. Until then, please use 2.2.12 that doesn't have this issue.
+
+* Enabling ``wsrep_desync`` (from previous ``OFF`` state) will wait until previous ``wsrep_desync=OFF`` operation is completed.
+
```

Comment:
* Check Hrvoje note about package installation.

* Also talk to Hrvoje about the new upgrade issue that we discovered. He has filed a bug for it. Not sure if we need to explicitly mention it here but just incase then let me leave this note as a place holder.